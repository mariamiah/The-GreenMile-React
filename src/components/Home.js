import React from 'react';

const Home = () => {
    return (
        <div>
            <p> Welcome to greenmile </p>
        </div>
    );
};
export default Home;
