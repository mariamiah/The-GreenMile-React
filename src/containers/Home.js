/* eslint-disable camelcase */
/* eslint-disable react/require-default-props */
/* eslint-disable class-methods-use-this */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ToastContainer, toast } from 'react-toastify';
import LandingPage from '../components/LandingPage';
import LoginAction from '../actions/LoginAction';

const jwt = require('jsonwebtoken');
// eslint-disable-next-line react/prefer-stateless-function
export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.loginState.user) {
      this.handleSuccess(nextProps.loginState);
    }

    if (!nextProps.loginState.user) {
      this.handleErrors(nextProps);
    }
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    // eslint-disable-next-line react/destructuring-assignment
    this.props.LoginAction(this.state);
  }

  handleSuccess(loginState) {
    if (loginState.user.access_token) {
      toast.success(
        <p className="text-white">
          Login Successful
        </p>,
      );
      const { access_token } = loginState.user;
      const decoded = jwt.decode(access_token);
      if (decoded.identity.role[0] === 'Admin') {
        setTimeout(() => window.location.assign('/admindashboard'), 2000);
      } else {
        setTimeout(() => window.location.assign('/loaderDashboard'), 2000);
      }
    }
  }

  handleErrors(props) {
    if (props.loginState.error) {
      toast.error(
        <p>
          {props.loginState.error.message}
        </p>,
      );
    }
  }

  render() {
    return (
      <div>
        <LandingPage onChange={this.handleChange} onSubmit={this.handleSubmit} />
        <ToastContainer />
      </div>
    );
  }
}
Home.propTypes = {
  LoginAction: PropTypes.func,
};

const mapStateToProps = state => ({
  username: state.LoginReducer.username,
  password: state.LoginReducer.password,
  loginState: state.LoginReducer,
});
export default connect(mapStateToProps, { LoginAction })(Home);
