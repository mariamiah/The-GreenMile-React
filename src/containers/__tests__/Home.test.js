/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../Home';

describe('<Home />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle errors on failed registration', () => {
    const props = {
      loginState: {
        error: {
          message: 'error',
        },
      },
      LoginAction: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);
    wrapper.setProps({
      loginState: {
      },
    });
    wrapper.instance().handleErrors(props);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().props.loginState).toEqual({});
  });

  it('should handle change', () => {
    const props = {
      loginState: {
        error: {
          message: 'error',
        },
      },
      LoginAction: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);
    const event = {
      target: {
        name: 'mia',
      },
    };
    wrapper.instance().handleChange(event);
    expect(wrapper.instance()).toMatchSnapshot();
  });
  it('should handle submit', () => {
    const props = {
      loginState: {
        error: {
          message: 'error',
        },
      },
      LoginAction: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);
    const event = {
      target: {
        name: 'mia',
      },
      preventDefault: jest.fn(),
    };
    wrapper.instance().handleSubmit(event);
    expect(wrapper.instance()).toMatchSnapshot();
  });
  it('should handle success with admin token', () => {
    const props = {
      loginState: {
        user: {
          message: 'error',
        },
      },
      LoginAction: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);
    wrapper.setProps({
      loginState: {
        user: {
          token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiQWRtaW4iLCJleHAiOjE1NTIwMTM4MDcsInJvbGVzIjpbIkFkbWluIl19.FLwsnsZnQWBC3u18oxvTtMTAqht2crgH4O7W2JM9Ql8',
        },
      },
    });
    wrapper.instance().handleSuccess(props.loginState);
    expect(wrapper.instance()).toMatchSnapshot();
  });
  it('should handle success with attendant token', () => {
    const props = {
      loginState: {
        user: {
          message: 'error',
        },
      },
      LoginAction: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);
    wrapper.setProps({
      loginState: {
        user: {
          token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoibWltaSIsImV4cCI6MTU1MjAxNjg2NSwicm9sZXMiOlsiQXR0ZW5kYW50Il19._sriBnI7smmCy6B_vy20LwBGh0vOW-ze1QbBzV5BiL4',
        },
      },
    });
    wrapper.instance().handleSuccess(props.loginState);
    expect(wrapper.instance()).toMatchSnapshot();
  });
});
