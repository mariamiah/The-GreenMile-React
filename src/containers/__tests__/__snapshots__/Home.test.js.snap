// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should handle change 1`] = `
Home {
  "componentWillReceiveProps": [Function],
  "context": Object {},
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": Object {
    "LoginAction": [MockFunction],
    "loginState": Object {
      "error": Object {
        "message": "error",
      },
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "mia": undefined,
    "password": "",
    "username": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Home
        LoginAction={[MockFunction]}
        loginState={
          Object {
            "error": Object {
              "message": "error",
            },
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <LandingPage
          onChange={[Function]}
          onSubmit={[Function]}
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`<Home /> should handle errors on failed registration 1`] = `ShallowWrapper {}`;

exports[`<Home /> should handle submit 1`] = `
Home {
  "componentWillReceiveProps": [Function],
  "context": Object {},
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": Object {
    "LoginAction": [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "password": "",
            "username": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "loginState": Object {
      "error": Object {
        "message": "error",
      },
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "password": "",
    "username": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Home
        LoginAction={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "password": "",
                  "username": "",
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        loginState={
          Object {
            "error": Object {
              "message": "error",
            },
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <LandingPage
          onChange={[Function]}
          onSubmit={[Function]}
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`<Home /> should handle success with admin token 1`] = `
Home {
  "componentWillReceiveProps": [Function],
  "context": Object {},
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": Object {
    "LoginAction": [MockFunction],
    "loginState": Object {
      "user": Object {
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiQWRtaW4iLCJleHAiOjE1NTIwMTM4MDcsInJvbGVzIjpbIkFkbWluIl19.FLwsnsZnQWBC3u18oxvTtMTAqht2crgH4O7W2JM9Ql8",
      },
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "password": "",
    "username": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Home
        LoginAction={[MockFunction]}
        loginState={
          Object {
            "user": Object {
              "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiQWRtaW4iLCJleHAiOjE1NTIwMTM4MDcsInJvbGVzIjpbIkFkbWluIl19.FLwsnsZnQWBC3u18oxvTtMTAqht2crgH4O7W2JM9Ql8",
            },
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <LandingPage
          onChange={[Function]}
          onSubmit={[Function]}
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`<Home /> should handle success with attendant token 1`] = `
Home {
  "componentWillReceiveProps": [Function],
  "context": Object {},
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": Object {
    "LoginAction": [MockFunction],
    "loginState": Object {
      "user": Object {
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoibWltaSIsImV4cCI6MTU1MjAxNjg2NSwicm9sZXMiOlsiQXR0ZW5kYW50Il19._sriBnI7smmCy6B_vy20LwBGh0vOW-ze1QbBzV5BiL4",
      },
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "password": "",
    "username": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Home
        LoginAction={[MockFunction]}
        loginState={
          Object {
            "user": Object {
              "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoibWltaSIsImV4cCI6MTU1MjAxNjg2NSwicm9sZXMiOlsiQXR0ZW5kYW50Il19._sriBnI7smmCy6B_vy20LwBGh0vOW-ze1QbBzV5BiL4",
            },
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <LandingPage
          onChange={[Function]}
          onSubmit={[Function]}
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`<Home /> should render without crashing 1`] = `ShallowWrapper {}`;
